### DISCUSSION TITLE
Copilot + Inline Diagnostics features: attempting local merge of PRs #6417 and #6865

### DISCUSSION THREAD

**Discussion by johnnyutahh** [helix-editor/helix#8235] [Q&A]
@pascalkuthe and @AlexanderDickie -

I've local-merged and built `hx` (with `cargo install --path helix-term --locked`; [some merge-sha details here](https://gist.github.com/johnnyutahh/3a30cceaee5a34895c8ff5f33ed34298)) for PRs Inline Diagnostics #6417 (@pascalkuthe) and Copilot #6865 (@AlexanderDickie), after performing a very-minor-and-obvious edit for a resulting code-conflict (in `helix-term/src/ui/mod.rs`). I've not tested yet beyond running and immediately-exiting `hx`. (I'm a new Helix user.)

Questions:

1. Does it make "useability logical" sense to do this?  ie: try and get "Copilot" and "CodeLens" features in the same `hx`?  ie: does it make "sense" to try to have both of these things/features in the same `hx`?
2. If "yes": do you see any obvious technical problems with attempting this, besides the fact this code state is completely untested (I'm willing to take on that mantle, at least to start)?
3. Is there anything else I should be aware of?  (eg: I'm a newbie Copilot and newbie-CodeLens user, haven' used these features in any context/IDE... so I have a pretty tall learning curve ahead of me... but please set that aside for the moment. 🤓)

ps. I'm completely new to this community, so apologies if there's any missteps I'm making with this inquiry.

**Comment by pascalkuthe** [comment]
I haven't had time to get back to the inline diagnostic PR yet, there are many things that are broken about it (diagnostics sometimes disappear, there are some crashes, ...) so you need to decide for yourself if that is a good idea. Can't comment on the copilot PR. I doubt there are a huge amount of conflicts.

**Comment by johnnyutahh** [comment]
Thanks pascalkuthe for your super-timely reply, quite helpful.

I'll await word from @AlexanderDickie (if he has time to comment; not required, of course) regarding his perspective from the Copilot-#6865 view of things.

**Reply by AlexanderDickie** [reply]
Yeh they will work together fine. Funny that you mention this because I've been experimenting with @pascalkuthe 's virtual text api to render copilot completions which should really improve my copilot fork.

**Reply by johnnyutahh** [reply]
Sounds great, I look forward to future copilot-fork updates.

Many thanks for all your (and pascalkuthe's) efforts!

**Reply by johnnyutahh** [reply]
For my reference: it appears this virtual-text-api feature merged to `master` in early 2023 (can someone pls correct me if I'm reading this wrongly?):

https://github.com/helix-editor/helix/pull/5420
